logic/suggestions/handToss_test.go

```diff
@@ -0,0 +1,202 @@
+package suggestions
+
+import (
+	"testing"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/network"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var _ model.TossStats = (*testingTossStats)(nil)
+
+type testingTossStats struct {
+	min    int
+	avg    float64
+	median float64
+	max    int
+}
+
+func (ts *testingTossStats) Min() int {
+	return ts.min
+}
+
+func (ts *testingTossStats) Median() float64 {
+	return ts.median
+}
+
+func (ts *testingTossStats) Avg() float64 {
+	return ts.avg
+}
+
+func (ts *testingTossStats) Max() int {
+	return ts.max
+}
+
+func TestGetStatsForHand(t *testing.T) {
+	tests := []struct {
+		desc         string
+		hand         []string
+		tossed       []string
+		expHandStats *testingTossStats
+		expCribStats *testingTossStats
+	}{{
+		desc:   `low pointer`,
+		hand:   []string{`AH`, `QH`, `JC`, `9D`},
+		tossed: []string{`10S`, `KS`},
+		expHandStats: &testingTossStats{
+			avg:    2.282608695652174,
+			median: 2,
+			max:    7,
+		},
+		expCribStats: &testingTossStats{
+			avg:    3.527975406236276,
+			median: 2,
+			max:    20,
+		},
+	}, {
+		desc:   `max pointer`,
+		hand:   []string{`5h`, `jd`, `5C`, `5s`},
+		tossed: []string{`4d`, `6d`},
+		expHandStats: &testingTossStats{
+			min:    14,
+			avg:    16.608695652173914,
+			median: 14,
+			max:    29,
+		},
+		expCribStats: &testingTossStats{
+			avg:    3.9596179183135707,
+			median: 4,
+			max:    24,
+		},
+	}}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.desc, func(t *testing.T) {
+			actHand, actCrib := getStatsForHand(
+				network.ModelCardsFromStrings(tc.hand...),
+				network.ModelCardsFromStrings(tc.tossed...),
+			)
+			assert.Equal(t, tc.expHandStats.Min(), actHand.Min())
+			assert.Equal(t, tc.expHandStats.Avg(), actHand.Avg())
```

Comment:
:question: Why is it preferable to test exact equality of two floats here?

Reply:
I don't typically test floats; so I'll update